sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -638,3 +641,66 @@ abstract class TypedImperativeAggregate[T] extends ImperativeAggregate {
     buffer.get(offset, anyObjectType).asInstanceOf[T]
   }
 }
+
+/**
+ * A special TypedImperativeAggregate that uses `OpenHashMap[AnyRef, Long]` as internal
+ * aggregation buffer.
+ */
+abstract class AggregationBuffer extends TypedImperativeAggregate[OpenHashMap[AnyRef, Long]] {
```

Comment:
`AggregationBuffer` is really a bad name. How about `TypedAggregateWithHashMapAsBuffer`?

Reply:
OK